niceday-api/service/UserTrackersService.js

```diff
@@ -21,3 +22,30 @@ exports.setUserTrackerStatuses = (req, body) => new Promise((resolve, reject) =>
       reject(err);
     });
 });
+
+/**
+ * Get smoking tracker data.
+ * @param req - The node.js express request object
+ * @param body - The node.js express body object.
+ * */
+exports.getSmokingTrackerData = (req) => new Promise((resolve, reject) => {
+  const { startTime, endTime } = req.query;
+  const { userId } = req.openapi.pathParams;
+  trackingSdk.init(req.app.get('token'), SenseServer.Alpha);
+  trackingSdk.getSensorResolved({
+    sensorName: ['tracker_smoking'],
+    startTime: new Date(startTime).toISOString(),
+    endTime: new Date(endTime).toISOString(),
+    userId,
+  })
+    .then((result) => {
+      // Retrieve nested data that is about cigarettes
+      const cigaretteData = result.map((c) => c.value.measures.measureCigarettes.sensorData);
```

Comment:
This is safe because the schema has ensured the result indeed has this nested structure, right?

Reply:
No this could break if what we get from niceday changes structure. I think indeed this is the point where we should maybe validate things? On the other hand, if this breaks it would be quite easy to find out that it breaks here. Maybe we should at some point just start logging errors.

Reply:
Or we should have more integration tests. Let's discuss it (or have a pointer for discussion and then park it for later investigation).

Reply:
I updated https://github.com/PerfectFit-project/virtual-coach-issues/issues/200 to continue the discussion at a later point.